scripts/update-chart-version.sh

```diff
@@ -0,0 +1,140 @@
+#!/usr/bin/env bash
+
+# Write output to error output stream.
+echo_stderr() {
+  echo -e "${1}" >&2
+}
+
+die()
+{
+  local _return="${2:-1}"
+  echo_stderr "$1"
+  exit "${_return}"
+}
+
+help() {
+  cat <<EOF
+Usage: $(basename "$0") [OPTIONS]
+
+Options:
+  --branch <branch name>                    Name of the target branch.
+  --tag <tag>                               Release tag.
+
+Examples:
+  $(basename "$0") --branch release/x.y
+EOF
+}
+
+# yq-go eats up blank lines
+# this function gets around that using diff with --ignore-blank-lines
+yq_ibl()
+{
+  set +e
+  diff_out=$(diff -B <(yq '.' "$2") <(yq "$1" "$2"))
+  error=$?
+  if [ "$error" != "0" ] && [ "$error" != "1" ]; then
+    exit "$error"
+  fi
+  if [ -n "$diff_out" ]; then
+    echo "$diff_out" | patch --quiet --no-backup-if-mismatch "$2" -
+  fi
+  set -euo pipefail
+}
+
+create_version_from_release_branch() {
+  if [[ "$BRANCH_NAME" =~ ^(release/[0-9]+\.[0-9]+)$ ]]; then
+    local EXTRACTED_VERSION=$(echo "$BRANCH_NAME" | grep -oP '(?<=release/)\d+\.\d+')
+    CURRENT_CHART_VERSION=$(yq e '.version' "$CHART_YAML")
+    if [[ "$CURRENT_CHART_VERSION" == *"-develop" ]]; then
+      VERSION="${EXTRACTED_VERSION}.0-prerelease"
+    elif [[ "$CURRENT_CHART_VERSION" == *"-prerelease" ]]; then
+      VERSION=$CURRENT_CHART_VERSION
```

Comment:
here we should leave it unset?

Reply:
No, I have made the scripts to fail if it's unset.